frontend/src/modules/entitydetails/components/EntityDetails.test.js

```diff
@@ -107,6 +117,45 @@ describe('<EntityDetailsBase>', () => {
         expect(wrapper.text()).toContain('Metadata');
         expect(wrapper.text()).toContain('Editor');
     });
+
+    it('shows failed checks for approved (or fuzzy) translations with errors or warnings', () => {
+        const wrapper = createShallowEntityDetails();
+
+        wrapper.setProps({
+            pluralForm: -1,
+            selectedEntity: {
+                pk: 2,
+                original: 'something',
+                translation: [{
+                    approved: true,
+                    string: 'quelque chose',
+                    errors: ['Error1'],
+                }],
+            },
+        });
+
+        expect(editor.actions.updateFailedChecks.calledOnce).toBeTruthy();
+        expect(editor.actions.resetFailedChecks.calledOnce).toBeFalsy();
+    });
+
+    it('hides failed checks for approved (or fuzzy) translations without errors or warnings', () => {
+        const wrapper = createShallowEntityDetails();
+
+        wrapper.setProps({
+            pluralForm: -1,
+            selectedEntity: {
+                pk: 2,
+                original: 'something',
+                translation: [{
+                    approved: true,
+                    string: 'quelque chose',
+                }],
+            },
+        });
+
+        expect(editor.actions.updateFailedChecks.calledOnce).toBeFalsy();
+        expect(editor.actions.resetFailedChecks.calledOnce).toBeTruthy();
```

Comment:
Have you checked that the method that calls this action is actually called? I usually run those kinds of debugs with `console.log`, and make sure that the correct path gets ran.

Reply:
Thanks for the debugging help. :) Indeed, there was a problem with the code and I fixed it.

But I'm only **almost** there... For some reason `editor.actions.updateFailedChecks.calledOnce` is still `truthy`. I thought count should be reset in `afterAll`?

Reply:
Ah, `afterAll` only runs after all tests have run. You want to use `afterEach` as well, see: https://github.com/mozilla/pontoon/blob/master/frontend/src/modules/editor/components/Editor.test.js#L66